miniapps/meshing/polar-nc.cpp

```diff
@@ -0,0 +1,558 @@
+
+      Array<int> dofs;
+      MFEM_ASSERT(params.Size() == mesh->GetNE(), "");
+
+      for (int i = 0; i < mesh->GetNE(); i++)
+      {
+         const Params3 &par = params[i];
+         const IntegrationRule &ir = fes->GetFE(i)->GetNodes();
+         Geometry::Type geom = mesh->GetElementBaseGeometry(i);
+         fes->GetElementDofs(i, dofs);
+
+         for (int j = 0; j < dofs.Size(); j++)
+         {
+            const IntegrationPoint &ip = ir[j];
+
+            double u, v, w, r;
+            if (geom == Geometry::PRISM)
+            {
+               double l1 = 1.0 - ip.x - ip.y;
+               double l2 = ip.x, l3 = ip.y;
+               u = l1 * par.u1 + l2 * par.u2 + l3 * par.u3;
+               v = l1 * par.v1 + l2 * par.v2 + l3 * par.v3;
+               w = 1.0 - u - v;
+               r = par.r + ip.z * par.dr;
+            }
+            else
+            {
+               u = ip.x * par.u1 + ip.y * par.u2 + ip.z * par.u3;
+               v = ip.x * par.v1 + ip.y * par.v2 + ip.z * par.v3;
+               double rr = ip.x + ip.y + ip.z;
+               if (std::abs(rr) < 1e-12) { continue; }
+               w = rr - u - v;
+               r = par.r + rr * par.dr;
+            }
+
+            double q = r / sqrt(u*u + v*v + w*w);
+            (*nodes)(fes->DofToVDof(dofs[j], 0)) = u*q;
+            (*nodes)(fes->DofToVDof(dofs[j], 1)) = v*q;
+            (*nodes)(fes->DofToVDof(dofs[j], 2)) = w*q;
+         }
+      }
+
+      nodes->RestrictConforming();
+   }
+
+   return mesh;
+}
+
+
+int main(int argc, char *argv[])
+{
+   int dim = 2;
+   double radius = 1.0;
+   int nsteps = 10;
+   double angle = 90;
+   double aspect = 1.0;
+   int order = 2;
+   bool sfc = true;
+   bool visualization = true;
+
+   // parse command line
+   OptionsParser args(argc, argv);
+   args.AddOption(&dim, "-d", "--dim", "Mesh dimension (2 or 3).");
+   args.AddOption(&radius, "-r", "--radius", "Radius of the domain.");
+   args.AddOption(&nsteps, "-n", "--nsteps",
+                  "Number of elements along the radial direction");
+   args.AddOption(&aspect, "-a", "--aspect",
+                  "Target aspect ratio of the elements.");
+   args.AddOption(&angle, "-phi", "--phi", "Angular range (2D only).");
+   args.AddOption(&order, "-o", "--order",
+                  "Polynomial degree of mesh curvature.");
+   args.AddOption(&sfc, "-sfc", "--sfc", "-no-sfc", "--no-sfc",
+                  "Try to order elements along a space-filling curve.");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return EXIT_FAILURE;
+   }
+   args.PrintOptions(cout);
+
+   // validate options
+   MFEM_VERIFY(dim >= 2 && dim <= 3, "");
+   MFEM_VERIFY(angle > 0 && angle < 360, "");
+   MFEM_VERIFY(nsteps > 0, "");
```

Comment:
Also `radius > 0` and `aspect > 0` (negative aspect ratio actually produces interesting meshes).

Reply:
Added, thanks.